static/app/views/onboarding/targetedOnboarding/setupDocs.tsx

```diff
@@ -0,0 +1,268 @@
+      setLoadedPlatform(project.platform);
+      setHasError(false);
+    } catch (error) {
+      setHasError(error);
+      throw error;
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, [project]);
+
+  // TODO: add better error handling logic
+  if (!project && subStep === 'project') {
+    return null;
+  }
+
+  const setNewProject = (newProjectId: string) => {
+    const searchParams = new URLSearchParams({
+      project_id: newProjectId,
+    });
+    browserHistory.push(`${window.location.pathname}?${searchParams}`);
+  };
+
+  const missingExampleWarning = () => {
+    const missingExample =
+      platformDocs && platformDocs.html.includes(INCOMPLETE_DOC_FLAG);
+
+    if (!missingExample) {
+      return null;
+    }
+
+    return (
+      <Alert type="warning" icon={<IconInfo size="md" />}>
+        {tct(
+          `Looks like this getting started example is still undergoing some
+           work and doesn't include an example for triggering an event quite
+           yet. If you have trouble sending your first event be sure to consult
+           the [docsLink:full documentation] for [platform].`,
+          {
+            docsLink: <ExternalLink href={platformDocs?.link} />,
+            platform: platforms.find(p => p.id === loadedPlatform)?.name,
+          }
+        )}
+      </Alert>
+    );
+  };
+
+  const docs = platformDocs !== null && (
+    <DocsWrapper key={platformDocs.html}>
+      <Content dangerouslySetInnerHTML={{__html: platformDocs.html}} />
+      {missingExampleWarning()}
+      {project && (
+        <FirstEventFooter
+          project={project}
+          organization={organization}
+          isLast={projectIndex === projects.length - 1}
+          hasFirstEvent={checkProjectHasFirstEvent(project)}
+          onClickSetupLater={() => {
+            // TODO: analytics
+            const nextProject = projects.find(
+              (p, index) => !p.firstEvent && index > projectIndex
+            );
+            if (!nextProject) {
+              // TODO: integrations
+              browserHistory.push('/');
+              return;
+            }
+            setNewProject(nextProject.id);
+          }}
+          handleFirstIssueReceived={() => {
+            const newHasFirstEventMap = {...hasFirstEventMap, [project.id]: true};
+            setHasFirstEventMap(newHasFirstEventMap);
+          }}
+        />
+      )}
+    </DocsWrapper>
+  );
+
+  const loadingError = (
+    <LoadingError
+      message={t('Failed to load documentation for the %s platform.', platform)}
+      onRetry={fetchData}
+    />
+  );
+
+  const testOnlyAlert = (
+    <Alert type="warning">
+      Platform documentation is not rendered in for tests in CI
+    </Alert>
+  );
+
+  return (
+    <Wrapper>
+      <TargetedOnboardingSidebar
+        activeProject={project}
+        {...{checkProjectHasFirstEvent, setNewProject}}
+      />
+      <MainContent>
+        <FullIntroduction currentPlatform={currentPlatform} />
```

Comment:
The transition here seems to be really hard, because when currentPlatform changes the content will disappear. This is probably not a problem before, but now when you switch between tabs the page will flash and the layout changes completely. Can you make it so that the docs doesn't disappear but instead retain the previous rendering instead, when currentPlatform changes?

Reply:
@Neo-Zhixing hmmm...but couldn't that cause a state where the selected project doesn't match the docs?

Reply:
@Neo-Zhixing ok, I updated it so we maintain the old instructions